### PR TITLE
Add support for <rule>

### PR DESCRIPTION
**Pull request by PiotrKozlowski**
Waiting for ~~icons and~~ changes to https://github.com/openstax-poland/cnx-designer/pull/86

To do:
- [x] ~~Solve problems with counters~~ it was related to counters definitions in cnx-designer
- [x] Update `cnx-designer`

There is new button:
obraz

When clicked new Rule will be created:
obraz

Rule have new tools:
obraz

There can be multiple statements, proofs and examples:
obraz

Title can be inserted with `+` from Insert tools:
obraz

When you click to insert `reference` then Rules should be properly listed:
obraz

### CODE REVIEW

File: src/@types/cnx-designer/index.d.ts [hunk 1]
```diff
@@ -77,6 +77,8 @@ declare module 'cnx-designer' {
     alt: string
   }
 
+  export type RuleChildType = 'example' | 'proof' | 'statement'
```
Comment (aiwenar) [inline src/@types/cnx-designer/index.d.ts]: This type isn't used anywhere.

Comment (PiotrKozlowski) [reply src/@types/cnx-designer/index.d.ts]: This is file with types for `cnx-designer` and if it is exported by `cnx-designer` then it should be declared even if it is not used in the code

**Comment by PiotrKozlowski** [comment]
> Adding titles to rules is not intuitive (after testing UI I thought it's impossible and only found out how to do it when I reviewed the code).

This is how it works also for `adminition` and `quatation` - in those you can also select `title` from upper menu, but in `rule` you can't because when you create it first element is `Statement` which does not allow `title`

> It's not possible to change rule's type.

Lets make separate issue for this: https://github.com/openstax-poland/adaptarr-front/issues/258

> What is LinkBox and why is it added in this PR?

Probably added this by mistake, I'll fix it after rebase

File: src/@types/cnx-designer/index.d.ts (hunk 1 above)
Comment (aiwenar) [reply src/@types/cnx-designer/index.d.ts]: What does it correspond to in `cnx-designer`?

Comment (PiotrKozlowski) [reply src/@types/cnx-designer/index.d.ts]: Oh, it is actually not exported, sry

**Comment by MartaSztuka** [comment]
tested.
Screenshot from 2020-01-14 12-39-02
Screenshot from 2020-01-14 12-38-49
Screenshot from 2020-01-14 12-38-31

**Pull request merged**